R/dplyr.R

```diff
@@ -261,17 +261,36 @@ prepare_join <- function(x, y, by, selected, suffix, copy, disambiguate = TRUE)
   if (quo_is_null(select_quo)) {
     selected <- tidyselect::vars_select(all_cols_y, everything())
   } else {
+    # FIXME: should we complain, if the user wishes to rename two or more columns to the same name?
```

Comment:
Is this the responsibility of tidyselect?

Reply:
`tidyselect::vars_select()` allows this:

``` r
dplyr::select(tibble::as_tibble(iris), name = Species, name = Sepal.Length)
#> # A tibble: 150 x 2
#>    name    name
#>    <fct>  <dbl>
#>  1 setosa   5.1
#>  2 setosa   4.9
#>  3 setosa   4.7
#>  4 setosa   4.6
#>  5 setosa   5  
#>  6 setosa   5.4
#>  7 setosa   4.6
#>  8 setosa   5  
#>  9 setosa   4.4
#> 10 setosa   4.9
#> # … with 140 more rows
```

<sup>Created on 2019-11-14 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>